client/src/pages/admin/MergeLocations.js

```diff
@@ -0,0 +1,516 @@
+      <AdvancedSingleSelect
+        fieldName="location"
+        placeholder="Select location to merge..."
+        value={location}
+        overlayColumns={["Name"]}
+        overlayRenderRow={LocationOverlayRow}
+        filterDefs={locationFilters}
+        objectType={Location}
+        fields={selectFields}
+        valueKey="name"
+        addon={LOCATIONS_ICON}
+        onChange={setLocation}
+      />
+      {location && (
+        <>
+          <LocationField
+            label="Name"
+            value={location.name}
+            align={reversed ? "right" : "left"}
+            reversed={reversed}
+            action={getActionButton(() => setFieldValue("name", location.name))}
+          />
+          <LocationField
+            label="Status"
+            value={location.status}
+            align={reversed ? "right" : "left"}
+            reversed={reversed}
+            action={getActionButton(() =>
+              setFieldValue("status", location.status)
+            )}
+          />
+          <LocationField
+            label="Location"
+            value={<GeoLocation lat={location.lat} lng={location.lng} />}
+            align={reversed ? "right" : "left"}
+            reversed={reversed}
+            action={getActionButton(() => {
+              setFieldValue("lat", location.lat)
+              setFieldValue("lng", location.lng)
+            })}
+          />
+          {map}
+          <div className={"merge-loc-fset" + (reversed ? " reversed" : "")}>
+            <ApprovalSteps
+              type={PLANNING_APPROVAL}
+              steps={location.planningApprovalSteps}
+              fieldSetAction={getActionButton(() =>
+                setFieldValue(
+                  "planningApprovalSteps",
+                  location.planningApprovalSteps
+                )
+              )}
+            />
+          </div>
+
+          <div className={"merge-loc-fset" + (reversed ? " reversed" : "")}>
+            <ApprovalSteps
+              type={PUBLICATION_APPROVAL}
+              steps={location.approvalSteps}
+              fieldSetAction={getActionButton(() =>
+                setFieldValue("approvalSteps", location.approvalSteps)
+              )}
+            />
+          </div>
+        </>
+      )}
+    </div>
+  )
+
+  function getActionButton(onClickAction) {
+    return (
+      <small>
+        <Button
+          icon={reversed ? "double-chevron-left" : ""}
+          rightIcon={reversed ? "" : "double-chevron-right"}
+          intent="primary"
+          text="Use value"
+          onClick={onClickAction}
+        />
+      </small>
+    )
+  }
+}
+
+LocationColumn.propTypes = {
+  location: PropTypes.instanceOf(Location),
+  setLocation: PropTypes.func.isRequired,
+  setFieldValue: PropTypes.func.isRequired,
+  reversed: PropTypes.bool,
+  style: PropTypes.object
+}
+
+LocationColumn.defaultProps = {
+  reversed: false
+}
+
+const LocationField = ({ label, value, align, reversed, action }) => {
```

Comment:
It seems like we are reinventing what a Field is, is a LocationField needed?

Reply:
`LocationField` mainly exists for avoiding code duplication. Also it encapsulates minor logic to have different layout for a field (aligning text, displaying button on right or left). It is used 6 times in `MergeLocations` and without it following code will be duplicated 6 times.
```
    <div
      style={{
        borderBottom: "1px solid #CCCCCC",
        display: "flex",
        flexDirection: reversed ? "row-reverse" : "row",
        justifyContent: "space-between",
        alignItems: "center",
        padding: "8px 0"
      }}
    >
      <div style={{ flex: "1 1 auto" }}>
        <div
          style={{
            fontWeight: "bold",
            textAlign: align || "center",
            textDecoration: "underline"
          }}
        >
          {label}
        </div>
        <div style={{ textAlign: align || "center" }}>{value}</div>
      </div>
      {action}
    </div>
```